internal/website/content/howto/pubsub/subscribe.md

```diff
@@ -0,0 +1,504 @@
+---
+title: "Subscribe to a Topic's Messages"
+date: 2019-03-26T09:44:33-07:00
+draft: true
+weight: 2
+---
+
+Subscribing to messages on a topic with the Go CDK takes two steps:
+
+1. Open the subscription with the Pub/Sub provider of your choice (once per subscription).
+2. Receive messages from the topic.
+
+The second step is the same across all providers because the first step
+creates a value of the portable [`*pubsub.Subscription`][] type. A simple
+subscriber that operates on messages serially looks like this:
+
+```go
+func main() {
+    // Parse configuration and open subscription (shutdown deferred).
+    // ...
+
+    // ... and then loop on received messages.
+    for {
+        msg, err := subscription.Receive(ctx)
+        if err != nil {
+            // Errors from Receive indicate that Receive will no longer succeed.
+            log.Printf("Receiving message: %v", err)
+            break
+        }
+        // TODO(light): Is this reasonable ack behavior, @jba?
+        if err := handle(ctx, msg); err != nil {
+            log.Printf("Handling message: %v", err)
+        } else {
+            msg.Ack()
+        }
+    }
+}
+
+func handle(ctx context.Context, msg *pubsub.Message) error {
+    // Do work based on the message.
+    fmt.Printf("Got message: %q\n", msg.Data)
+    return nil
+}
+```
+
+If you want your subscriber to operate on the incoming messages concurrently,
+you can start multiple goroutines:
+
+```go
+func main() {
+    // Parse configuration and open subscription (shutdown deferred).
+    // ...
+
+    // ... and then loop on received messages. We can use a channel as a
+    // semaphore to limit how many goroutines we have active at a time as well
+    // as wait on the goroutines to finish before exiting.
+    const maxHandlers = 10
+    sem := make(chan struct{}, maxHandlers)
+recvLoop:
+    for {
+        msg, err := subscription.Receive(ctx)
+        if err != nil {
+            // Errors from Receive indicate that Receive will no longer succeed.
+            log.Printf("Receiving message: %v", err)
+            break
+        }
+
+        // Wait if there are too many active handle goroutines and
+        // acquire the semaphore.
+        select {
+        case sem <- struct{}{}:
+        case <-ctx.Done():
+            break recvLoop
+        }
+
+        // Handle the message in a new goroutine.
+        go func() {
+            // TODO(light): Is this reasonable ack behavior, @jba?
+            if err := handle(ctx, msg); err != nil {
+                log.Printf("Handling message: %v", err)
+            } else {
+                msg.Ack()
+            }
+            // Release the semaphore.
+            <-sem
```

Comment:
Put `defer func() { <-sem }()` at the top of this closure instead. Doesn't matter here but a good thing to encourage.

Reply:
Done.